src/components/organisms/Settings/Project/Dataset/hooks.ts

```diff
@@ -27,15 +27,36 @@ export default (projectId: string) => {
   });
 
   const sceneId = sceneData?.scene?.id;
+  const dataSetPerPage = 20;
 
-  const { data } = useGetDatasetSchemasQuery({
-    variables: { projectId: projectId ?? "", first: 100 },
+  const { data, fetchMore, loading, networkStatus } = useDatasetsListQuery({
+    variables: { sceneId: sceneId ?? "", first: dataSetPerPage },
     skip: !projectId,
+    notifyOnNetworkStatusChange: true,
   });
 
-  const nodes = data?.scene?.datasetSchemas.nodes ?? [];
+  const nodes = data?.datasetSchemas.edges.map(e => e.node) ?? [];
 
   const datasetSchemas = nodes.filter(Boolean) as DatasetSchemas;
+
+  const hasMoreDataSets =
+    data?.datasetSchemas?.pageInfo.hasNextPage || data?.datasetSchemas?.pageInfo.hasPreviousPage;
+
+  const isRefetchingDataSets = networkStatus === 3;
+
+  const getMoreDataSets = useCallback(() => {
+    if (hasMoreDataSets) {
+      fetchMore({
+        variables: {
+          after: data?.datasetSchemas?.pageInfo.endCursor,
+        },
+        updateQuery: (prev, { fetchMoreResult }) => {
+          if (!fetchMoreResult) return prev;
+          return fetchMoreResult;
+        },
+      });
+    }
+  }, [data?.datasetSchemas?.pageInfo, fetchMore, hasMoreDataSets]);
   const client = useApolloClient();
```

Comment:
Spacing please. Also, if it seems more appropriate to you to move this declaration above with useIntl and the state, please do.